src/main/java/io/kokuwa/keycloak/metrics/store/MicrometerEventStoreProviderFactory.java

```diff
@@ -0,0 +1,51 @@
+package io.kokuwa.keycloak.metrics.store;
+
+import io.kokuwa.keycloak.metrics.CommunityProfiles;
+import io.kokuwa.keycloak.metrics.event.MetricsEventListenerFactory;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.connections.jpa.JpaConnectionProvider;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.jpa.JpaEventStoreProviderFactory;
+import org.keycloak.models.KeycloakSession;
+
+import java.util.stream.Collectors;
+
+public class MicrometerEventStoreProviderFactory extends JpaEventStoreProviderFactory {
+    private static final Logger log = Logger.getLogger(MicrometerEventStoreProviderFactory.class);
+
+    public static final String ID = "jpa"; // Override default event store provider
+    private boolean replaceIds;
+
+    @Override
+    public EventStoreProvider create(KeycloakSession session) {
+        //removeMetricsEventsListenerIfExists(session);
+
+        JpaConnectionProvider connection = session.getProvider(JpaConnectionProvider.class);
+        return new MicrometerEventStoreProvider(session, connection.getEntityManager(), replaceIds);
+    }
+
+    public void init(Config.Scope config) {
+        replaceIds = "true".equals(System.getenv().getOrDefault("KC_METRICS_EVENT_REPLACE_IDS", "true"));
+        log.info(replaceIds ? "Configured with model names." : "Configured with model ids.");
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    // we need to remove 'metrics-listener' if already exists in realm, and KC_COMMUNITY_EVENTS_METRICS_ENABLED=true otherwise the keycloak_1  | 2024-06-22 06:37:29,059 ERROR [org.keycloak.services] (executor-thread-3) KC-SERVICES0083: Event listener 'metrics-listener' registered, but provider not found
```

Comment:
@xgp if you have time please check this comment

Reply:
It's "not found" because there is no `src/main/resources/META-INF/services/org.keycloak.events.EventListenerProviderFactory` services file

Reply:
Correct. But I think is what we want, right? if KC_COMMUNITY_EVENTS_METRICS_ENABLED=true a keycloak admin should not be able to add the 'metrics-listener' in the events in order to have duplicate count. This is why I've left the MetricsEventListenerFactory to implement EnvironmentDependentProviderFactory